source/simulator/assembly.cc

```diff
@@ -1624,14 +1644,37 @@ namespace aspect
     // TODO: Compute artificial viscosity once per timestep instead of each time
     // temperature system is assembled (as this might happen more than once per
     // timestep for iterative solvers)
-    const double nu
-      = compute_viscosity (scratch,
+
+    double nu = 0;
+	int count_additions = 0;
+	double nu_additions = 0.0;
+    if (use_smoothing == false)
+    {
+      nu = compute_viscosity (scratch,
```

Comment:
I said this above but, I would still need the boolean "use_smoothing" inside this function because when it gets to "nu" I don't want it to do what it normally does, I want it to use what is inside "viscosity_per_cell". I can make the averaging (or taking max) happen outside of this though.

Unless you have a better idea to do this?